modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,423 @@
+/**
+ * This module adds Sirdata provider to the real time data module
+ * The {@link module:modules/realTimeData} module is required
+ * The module will fetch segments (user-centric) and categories (page-centric) from Sirdata server
+ * The module will automatically handle user's privacy and choice in California (IAB TL CCPA Framework) and in Europe (IAB EU TCF FOR GDPR)
+ * @module modules/sirdataRtdProvider
+ * @requires module:modules/realTimeData
+ */
+import {getGlobal} from '../src/prebidGlobal.js';
+import * as utils from '../src/utils.js';
+import {submodule} from '../src/hook.js';
+import {ajax} from '../src/ajax.js';
+import findIndex from 'core-js-pure/features/array/find-index.js';
+import { getRefererInfo } from '../src/refererDetection.js';
+import { config } from '../src/config.js';
+
+/** @type {string} */
+const MODULE_NAME = 'realTimeData';
+const SUBMODULE_NAME = 'SirdataRTDModule';
+
+const _set = (obj, path, val, override) => {
```

Comment:
@nouchy you can probably utilize the utils.deepSetValue/mergeDeep functions respectively

Reply:
Done and has been pushed now... Thank you for this advice